core/contracts/oracle/implementation/EncryptedStore.sol

```diff
@@ -0,0 +1,59 @@
+pragma solidity ^0.5.0;
+
+/**
+ * @title Contract for storing keyed encrypted messages via the EVM.
+ * @dev This contract uses topic hashes as keys and can store a single arbitrary encrypted message per topic at any
+ * given time.
+ * There's technically nothing that requires the topics hashed or for the messages to be encrypted.
+ * This contract is built for the following specific use case:
+ * - The sender knows the topics ahead of time. This can either be communicated elsewhere or be implicit.
+ * - Only one message per topic is stored at any given time.
+ */
+contract EncryptedStore {
+    struct TopicData {
+        // The encrypted message.
+        bytes message;
+    }
+
+    struct Store {
+        // This maps from a hash to the data for this topic.
+        // Note: the hash is a hash of the "subject" or "topic" of the message.
+        mapping(bytes32 => TopicData) topics;
+    }
+
+    // Mapping from addresses to topic hashes to encrypted messages
+    mapping(address => Store) private stores;
+
+    /**
+     * @notice Gets the current stored message corresponding to `owner` and `topicHash`.
```

Comment:
nit: maybe replace owner with something else since owner is generally used to signify the owner of a contract, not a message? Not super opinionated though.

Reply:
@mrice32 check the latest contract, I greatly simplified the data structure

Reply:
SGTM.